crates/biome_js_analyze/src/lint/nursery/use_consistent_string_functions.rs

```diff
@@ -0,0 +1,189 @@
+    /// - [Differences between substring() and slice()](https:
+    /// //developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring#differences_between_substring_and_slice)
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// const foo = bar.trimLeft();
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const foo = bar.trimRight();
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// foo.substr(start, length);
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// foo.substring(indexStart, indexEnd);
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// const foo = bar.trimStart();
+    /// ```
+    ///
+    /// ```js
+    /// const foo = bar.trimEnd();
+    /// ```
+    ///
+    /// ```js
+    /// foo.slice(beginIndex, endIndex);
+    /// ```
+    ///
+    pub UseConsistentStringFunctions {
+        version: "next",
+        name: "useConsistentStringFunctions",
+        language: "js",
+        recommended: false,
+        fix_kind: FixKind::Safe,
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct UseConsistentStringFunctionsState {
+    member_name: TokenText,
+    span: TextRange,
+}
+
+impl Rule for UseConsistentStringFunctions {
+    type Query = Ast<JsCallExpression>;
+    type State = UseConsistentStringFunctionsState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let callee = node.callee().ok()?;
+        let expression = callee.as_js_static_member_expression()?;
+        let value_token = expression.member().ok()?.value_token().ok()?;
+        let string_function_name = value_token.text_trimmed();
+
+        match string_function_name {
+            "trimLeft" | "trimRight" | "substr" | "substring" => {
+                Some(UseConsistentStringFunctionsState {
+                    member_name: value_token.token_text_trimmed(),
+                    span: value_token.text_range(),
+                })
+            }
+            _ => None,
+        }
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let member_name = state.member_name.text();
+        let replaced_member_name = match member_name {
+            "trimLeft" => "trimStart()",
+            "trimRight" => "trimEnd()",
+            "substr" | "substring" => "slice()",
+            _ => return None,
+        };
+        let span = state.span;
+        let diagnostic_message = markup! {
+            "Avoid using "{member_name}" and consider using "{replaced_member_name}" instead."
+        }
+        .to_owned();
+        let note_message = if member_name == "substring" {
```

Comment:
@minht11 
There is a subtle differences between `substr` and `slice` too especially in the interpretation of the second argument and when start is greater than stop.
I would like to paste MDN documentation because it's simple and clear.

https://262.ecma-international.org/9.0/#sec-string.prototype.slice
https://rapd.wordpress.com/2007/07/12/javascript-substr-vs-substring/
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr#description